dart/common/NameManager.h

```diff
@@ -189,6 +227,12 @@ class NameManager
     return (mMap.find(_name) != mMap.end());
   }
 
+  /// Return true if the object is contained
+  bool hasObject(T _obj) const
+  {
+    return (mReverseMap.find(_obj) != mReverseMap.end());
+  }
+
   /// Get the number of the objects currently stored by the NameManager
   size_t getCount() const
   {
```

Comment:
I was extremely careful to ensure that mMap and mReverseMap are always in sync with each other (they always insert and erase corresponding entries at the same time). The only way it could be violated is if the class is inherited, and the derived class does something insane with them. We could prevent that by making those variables private instead of protected, if that's a concern.

That said, I wouldn't be opposed to having a debug mode check somewhere ensuring that all the entries are in sync, but I would question whether the increased runtime would be worth it.